main.go

```diff
@@ -0,0 +1,176 @@
+
+// parseListeners determines whether each listen address is IPv4 and IPv6 and
+// returns a slice of appropriate net.Addrs to listen on with TCP. It also
+// properly detects addresses which apply to "all interfaces" and adds the
+// address as both IPv4 and IPv6.
+func parseListeners(addrs []string) ([]net.Addr, error) {
+	netAddrs := make([]net.Addr, 0, len(addrs)*2)
+	for _, addr := range addrs {
+		host, _, err := net.SplitHostPort(addr)
+		if err != nil {
+			// Shouldn't happen due to already being normalized.
+			return nil, err
+		}
+
+		// Empty host or host of * on plan9 is both IPv4 and IPv6.
+		if host == "" || (host == "*" && runtime.GOOS == "plan9") {
+			netAddrs = append(netAddrs, simpleAddr{net: "tcp4", addr: addr})
+			netAddrs = append(netAddrs, simpleAddr{net: "tcp6", addr: addr})
+			continue
+		}
+
+		// Strip IPv6 zone id if present since net.ParseIP does not
+		// handle it.
+		zoneIndex := strings.LastIndex(host, "%")
+		if zoneIndex > 0 {
+			host = host[:zoneIndex]
+		}
+
+		// Parse the IP.
+		ip := net.ParseIP(host)
+		if ip == nil {
+			return nil, fmt.Errorf("'%s' is not a valid IP address", host)
+		}
+
+		// To4 returns nil when the IP is not an IPv4 address, so use
+		// this determine the address type.
+		if ip.To4() == nil {
+			netAddrs = append(netAddrs, simpleAddr{net: "tcp6", addr: addr})
+		} else {
+			netAddrs = append(netAddrs, simpleAddr{net: "tcp4", addr: addr})
+		}
+	}
+	return netAddrs, nil
+}
+
+// setupRPCListeners returns a slice of listeners that are configured for use
+// with the RPC server depending on the configuration settings for listen
+// addresses and TLS.
+func setupRPCListeners() ([]net.Listener, error) {
+	// Setup TLS if not disabled.
+	listenFunc := net.Listen
+
+	// TODO: check and discuss if TLS certificate needed
+	// if !cfg.DisableTLS {
+	// 	// Generate the TLS cert and key file if both don't already
+	// 	// exist.
+	// 	if !fileExists(cfg.RPCKey) && !fileExists(cfg.RPCCert) {
+	// 		err := genCertPair(cfg.RPCCert, cfg.RPCKey)
+	// 		if err != nil {
+	// 			return nil, err
+	// 		}
+	// 	}
+	// 	keypair, err := tls.LoadX509KeyPair(cfg.RPCCert, cfg.RPCKey)
+	// 	if err != nil {
+	// 		return nil, err
+	// 	}
+
+	// 	tlsConfig := tls.Config{
+	// 		Certificates: []tls.Certificate{keypair},
+	// 		MinVersion:   tls.VersionTLS12,
+	// 	}
+
+	// 	// Change the standard net.Listen function to the tls one.
+	// 	listenFunc = func(net string, laddr string) (net.Listener, error) {
+	// 		return tls.Listen(net, laddr, &tlsConfig)
+	// 	}
+	// }
+
+	// TODO: below addr and port should be moved to config file
+	addrs, err := net.LookupHost("localhost")
+	rpcPort := "8334"
+	if err != nil {
+		return nil, err
+	}
+	RPCListeners := make([]string, 0, len(addrs))
+	for _, addr := range addrs {
+		addr = net.JoinHostPort(addr, rpcPort)
+		RPCListeners = append(RPCListeners, addr)
+	}
+
+	netAddrs, err := parseListeners(RPCListeners)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
let's move this function to other file. `main.go` should be only about the CLI and calling other functions + helpers for CLI